### PR TITLE
topics: "style" category & entries

### PR DESCRIPTION
**Pull request by tleen**
Moved "reactive programming" to its own section along with
alternatives like OOP and functional. Added two test styles
test driven development and behavior driven development.

These all have a place in the exercises. "functional programming" will
hopefully replace a the higher-order functions topics and functional in dlang
and ocaml.

TDD is a lot of what Exercism does. I propose to limit it to the
exercise that benefit specifically from incremental development like
[sum-of-multiples](https://github.com/exercism/problem-specifications/tree/master/exercises/sum-of-multiples). That
being said, should a track use BDD that should be available in TOPICS as well.

See #884

### CODE REVIEW

File: TOPICS.txt [hunk 1]
```diff
@@ -63,11 +63,18 @@ Parsing
...
 Security
 Sorting
 Text formatting
 Transforming
+
+Programming Style
+---
+Behavior driven development
+Functional programming
+Object oriented programming
```
Comment (ErikSchierboom) [inline TOPICS.txt]: According to [Wikipedia](https://en.wikipedia.org/wiki/Object-oriented_programming), it is written with a dash: Object-oriented programming

File: TOPICS.txt [hunk 2]
```diff
@@ -63,11 +63,18 @@ Parsing
...
 Regular Expressions
 Searching
 Security
 Sorting
 Text formatting
 Transforming
+
+Programming Style
+---
+Behavior driven development
```
Comment (ErikSchierboom) [inline TOPICS.txt]: According to [Wikipedia](https://en.wikipedia.org/wiki/Behavior-driven_development),  it is written with a dash: Behavior-driven development

File: TOPICS.txt [hunk 3]
```diff
@@ -63,11 +63,18 @@ Parsing
...
+Functional programming
+Object oriented programming
+Reactive programming
+Test driven development
```
Comment (ErikSchierboom) [inline TOPICS.txt]: According to [Wikipedia](https://en.wikipedia.org/wiki/Test-driven_development), it is written with a dash: Test-driven development

**Comment by tleen** [comment]
That was an ugly merge. I didn't squash sorry :disappointed:.

**Pull request merged**